starky/src/linearhash.rs

```diff
@@ -95,6 +139,162 @@ impl LinearHash {
         }
         Ok(ElementDigest::<4>::new(&st))
     }
+
+    #[cfg(all(
+        target_feature = "avx512bw",
+        target_feature = "avx512cd",
+        target_feature = "avx512dq",
+        target_feature = "avx512f",
+        target_feature = "avx512vl"
+    ))]
+    pub fn hash_element_matrix(
+        &self,
+        vals: &[Vec<FGL>],
+        batch_size: usize,
+    ) -> Result<ElementDigest<4>> {
+        let mut flatvals: Vec<FGL> = vec![];
```

Comment:
Here we can opetimize this flatvals = vec![ ; vals.len() * vals[0].len()];  use rayon and slice::copy_from_slice to fill the flatvals.